kentik_api_library/kentik_api/public/device_label.py

```diff
@@ -16,56 +16,46 @@ class DeviceItem:
 # pylint: disable=too-many-instance-attributes
 
 
+@dataclass
 class DeviceLabel:
-    # pylint: disable=too-many-arguments
-    def __init__(
-        self,
-        name: str,
-        color: Optional[str] = None,
-        id: Optional[ID] = None,
-        user_id: Optional[ID] = None,
-        company_id: Optional[ID] = None,
-        devices: Optional[List[DeviceItem]] = None,
-        created_date: Optional[str] = None,
-        updated_date: Optional[str] = None,
-    ) -> None:
-        # read-write
-        self.name = name
-        self.color = color
-
-        # read-only
-        self._id = id
-        self._user_id = user_id
-        self._company_id = company_id
-        self._devices = devices
-        self._created_date = created_date
-        self._updated_date = updated_date
-
-    # pylint: enable=too-many-arguments
+    # read-write
+    name: str
+    color: str
+
+    # read-only
+    _devices: List[DeviceItem] = field(default_factory=list)
+    _id: ID = DEFAULT_ID
+    _user_id: Optional[ID] = DEFAULT_ID
+    _company_id: ID = DEFAULT_ID
+    _created_date: str = DEFAULT_DATE
+    _updated_date: str = DEFAULT_DATE
+
+    @classmethod
+    def new(cls, name: str, color: str):
+        return cls(name=name, color=color)
 
     @property
     def id(self) -> ID:
-        assert self._id is not None
         return self._id
 
     @property
     def user_id(self) -> Optional[ID]:
         return self._user_id
 
     @property
-    def company_id(self) -> Optional[ID]:
+    def company_id(self) -> ID:
         return self._company_id
 
     @property
     def devices(self) -> List[DeviceItem]:
         return [] if self._devices is None else self._devices
```

Comment:
Question: after changes to initializer isn't  `return self._devices` sufficient here?

Reply:
Resolved